sdk/tables/Azure.Data.Tables/tests/TableEntityTests.cs

```diff
@@ -246,5 +246,27 @@ public void TypeCoercionForNumericTypes()
             Assert.That(entity["Foo3"], Is.EqualTo(now));
             Assert.That(entity["Foo3"] is DateTime);
         }
+
+         [Test]
+        public void TypeCoercionForDateTimeTypes()
```

Comment:
Do we need a test for model mapping?

Reply:
I believe this only affects TableEntity not custom models.